cmd/dockerd/daemon.go

```diff
@@ -293,16 +294,24 @@ func newRouterOptions(config *config.Config, daemon *daemon.Daemon) (routerOptio
 		return opts, err
 	}
 
-	bb, err := buildbackend.NewBackend(daemon.ImageService(), manager, buildCache, buildkit)
+	bb, err := buildbackend.NewBackend(daemon.ImageService(), manager, buildCache, bk)
 	if err != nil {
 		return opts, errors.Wrap(err, "failed to create buildmanager")
 	}
-
+	var bv types.BuilderVersion
+	if v, ok := config.Features["buildkit"]; ok {
+		if v {
+			bv = types.BuilderBuildKit
+		} else {
+			bv = types.BuilderV1
```

Comment:
If the `buildkit` feature is not set in `daemon.json`, no `Builder-Version` header is set? I'd expect `features: {buildkit:false}}` and `{}` to be equivalent (i.e. in both cases buildkit is not active, so builder version is 1)

Reply:
@thaJeztah yes. So it doesn't matter.

Reply:
@thaJeztah We actually meant to make it distinguished, so `features: {buildkit:false}}` means `buildkit` is not allowed while `{}` (or not set in daemon.json)  means it's up to client to choose builder version. For header it is the same concept.

Reply:
Still confused by this; how would the client know that buildkit is available?

Reply:
@thaJeztah It's obvious that when the `features` is set in `daemon.json` client will know the version. If it is not set in the config file (i.e. header will not include `Builder-Version` either), `cli` will use the env var to determine whether to make buildkit request or otherwise use the old builder. Maybe the changes from cli can also help you - https://github.com/docker/cli/pull/1275

Reply:
Got it